EP-4: Statistics Screen:

LeonidVolohov opened the pull request:
- Update Constants.kt, CoronavirusApi.kt, CoronavirusService.kt files
  - Add fun provideStatisticsPresenter(schedulerProvider: SchedulerProviderImpl) to AppModule.kt
  - Add data class CountryResponse() for api data
  - Add statistics/... files for ui
  - Add icons for statistics screen
  - Create ui for statistics screen

LeonidVolohov committed:
EP-4: Statistics Screen:
  - Update Constants.kt, CoronavirusApi.kt, CoronavirusService.kt files
  - Add fun provideStatisticsPresenter(schedulerProvider: SchedulerProviderImpl) to AppModule.kt
  - Add data class CountryResponse() for api data
  - Add statistics/... files for ui
  - Add icons for statistics screen
  - Create ui for statistics screen

mordvintseva commented on .idea/jarRepositories.xml:
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Файл генерируется автоматически, можно не добавлять его в проект

mordvintseva commented on app/src/main/AndroidManifest.xml:
```diff
@@ -2,19 +2,21 @@
     package="com.example.epamcoronavirusmap">
 
-    <uses-permission android:name="android.permission.INTERNET"/>
+    <uses-permission android:name="android.permission.INTERNET" />
```
Правки в файле, вроде, ничего не меняют, можно откатить

Aliot-EI3 replied:
не надо откатывать. это верное форматирование

mordvintseva commented on app/src/main/java/com/example/epamcoronavirusmap/ui/countries/CountriesFragment.kt:
```diff
@@ -1,10 +1,10 @@
 package com.example.epamcoronavirusmap.ui.countries
 
 import android.os.Bundle
-import androidx.fragment.app.Fragment
```
Ненужные правки

Aliot-EI3 replied:
нужные, если импорт не используется - его надо удалять

mordvintseva commented on app/src/main/java/com/example/epamcoronavirusmap/ui/statistics/StatisticsRecyclerViewAdapter.kt:
```diff
@@ -0,0 +1,54 @@
+        private var mHealthyCasesView: TextView? = null
+
+        init {
+            mCountryNameView = itemView.findViewById(R.id.text_country_name)
```
Можно избавиться от множественного findViewById через kotlinx.android.synthetic

LeonidVolohov replied:
Исправил

mordvintseva reviewed (approved):
Всё хорошо, можно во время создания коммита не добавлять файлы, которые трогал во время работы, но на результат выполнения таски не влияют (вроде изменения стартового экрана, автоматически создаваемый jarRepositories.xml) :)

Aliot-EI3 commented on app/src/main/java/com/example/epamcoronavirusmap/ui/countries/CountryResponse.kt:
```diff
@@ -0,0 +1,27 @@
+)
+
+data class CountryInfo(
+    val _id: Int?,
```
нехороший нейминг. есть возможность называть поля нормально, и связывать их с пришедшими из апи

LeonidVolohov replied:
Удалил data class CountryInfo, т.к нигде не используется и не будет. У data class CountryResponse изменил нейминг на более очевидный.

Aliot-EI3 commented on app/src/main/java/com/example/epamcoronavirusmap/ui/statistics/StatisticsPresenter.kt:
```diff
@@ -0,0 +1,34 @@
+                    {
+                        it?.let {
+                            view?.displayPosts(it)
+                            Log.i(TAG, it.size.toString())
```
зачем используется  TAG?

LeonidVolohov replied:
Чтобы было проще отслеживать в Logcat`е.  Удалил

Aliot-EI3 commented on app/src/main/res/layout/activity_recycler_item.xml:
```diff
@@ -0,0 +1,178 @@
+            <TextView
+                android:id="@+id/text_country_name"
+                android:layout_width="match_parent"
+                android:layout_height="28dp"
```
все dp должны быть в отдельном файле и в xml на них ссылки

LeonidVolohov replied:
Исправил

Aliot-EI3 commented on app/src/main/res/layout/activity_recycler_item.xml:
```diff
@@ -0,0 +1,178 @@
+                android:layout_marginBottom="4dp"
+                android:textSize="20sp"
+                android:textStyle="italic|bold"
+                tools:text="Country name" />
```
захардкоженых текстов тоже не должно быть

LeonidVolohov replied:
Согласен, но tools:... используется же только на момент верстки, а в готовое приложение эта строка не уйдет же. Все остальные строки лежат в strings.xml

Aliot-EI3 commented on app/src/main/res/navigation/navigation.xml:
```diff
@@ -29,9 +29,10 @@
         android:name="com.example.epamcoronavirusmap.GoogleMapFragment"
         android:label="fragment_google_map"
-        tools:layout="@layout/fragment_google_map" />
+        tools:layout="@layout/fragment_google_map" /> -->
```
?

LeonidVolohov replied:
Студия ругается на то, что GoogleMapFragment отсутствует, поэтому я закомментировал этот фрагмент

LeonidVolohov committed:
EP-4: Statistics Screen:
  - Update activity_recycler_item.xml: add link to the values in dimens.xml file
  - Change naming in CountryResponse.kt
  - Delete TAG in StatisticsPresenter.kt
  - Change findViewByID() binding using Kotlin Android Extensions

mordvintseva committed:
Merge branch 'develop' into feature/EP-4-statistics-screen
